tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+            } elsif ($key == 'GINKGO_SKIP') {
+                $skip_test = "-ginkgo.skip=\"${$section}{$key}\"";
+            } elsif ($key == 'EXTRA_OPT') {
+                $extra_opt = ${$section}{$key};
+            }
+        }
+        $artifacts = "/tmp/artifacts/$section";
+        $junit_xml = "/tmp/artifacts/$section\_test.xml";
+        $test_log = "/tmp/artifacts/$section\_test.log";
+        script_run("virt-tests -ginkgo.focus=\"$go_test\" \
+                    $skip_test \
+                    -ginkgo.slowSpecThreshold 60 \
+                    -kubeconfig=/root/.kube/config \
+                    -kubectl-path=`which kubectl` \
+                    -virtctl-path=`which virtctl` \
+                    -installed-namespace=kubevirt \
+                    -deploy-testing-infra=false \
+                    -config=/usr/share/kube-virt/manifests/testing/default-config.json \
+                    -ginkgo.v=true \
+                    -test.v=true \
+                    -ginkgo.trace=true \
+                    -ginkgo.noisySkippings=false \
+                    -ginkgo.progress=true \
+                    -ginkgo.noColor \
+                    $extra_opt \
+                    -artifacts=$artifacts \
+                    -junit-output=$junit_xml \
+                    -test.testlogfile=$test_log");
+		
+        assert_script_run("sed -ie 's/Tests Suite/$section/g' $junit_xml");
+    }
+}
+
+sub generate_test_report {
+    my $result_path = "/tmp/artifacts";
+    my $build_xml = "./buildTestReports.xml";
+
+    script_run("cat > $build_xml <<__END
+<project name=\"genTestReport\" default=\"gen\" basedir=\".\">
+    <description>
+        Generate the HTML report from JUnit XML files
+    </description>
+    <target name=\"gen\">
+        <property name=\"genReportDir\" location=\"$result_path\"/>
+        <delete dir=\"$result_path/html\"/>
+        <mkdir dir=\"$result_path/html\"/>
+        <junitreport todir=\"$result_path\">
+            <fileset dir=\"$result_path\">
+                <include name=\"*_test.xml\" />
+            </fileset>
+            <report format=\"frames\" todir=\"$result_path/html\" />
+        </junitreport>
+    </target>
+</project>
+__END");
+
+    # Generate JUnit HTML aggregate test reports by ant
+    script_run("ant -buildfile $build_xml");
+}
+
+sub upload_results {
+    # Compress test logs as a tarball and upload to openqa job
+    my $test_artifacts = "/tmp/test_artifacts.tar.gz";
+    assert_script_run("tar czvf $test_artifacts /tmp/artifacts");
+    upload_logs $test_artifacts;
+
+    my $html_dir = "/tmp/artifacts/html";
+    opendir(DIR, $html_dir) or die "Can't open the directory.";
+    my @html_files = readdir DIR;
+    foreach (@html_files) {
+        upload_logs $_;
+    }
+}
+
+sub post_fail_hook {
+    # Getting container status, uploading test reports & error logs for analysis
+    select_console 'log-console';
+    upload_results();
+}
+
+1;
```

Comment:
Btw, in this PR, overall, there is no enough debug info in openqa job webui to let job reviewers quickly know the key test information. Can you add more record_info to indicate key test step start/end, and other key informations like tested kubevirt version, rke2 version, test result, etc?

Reply:
rke2 version:
- http://openqa.qa2.suse.asia/image/4a9/48b/.thumbs/0e486f830c211388ffbd6dad8f.png

kubevirt version:
- http://openqa.qa2.suse.asia/image/2b4/423/.thumbs/efc1b07ad4017b4ff01794d5cb.png
- http://openqa.qa2.suse.asia/tests/49681#step/kubevirt_tests_server/78http://openqa.qa2.suse.asia/tests/49681#step/kubevirt_tests_server/78